pkg/controllers/management/eks/eks_cluster_handler.go

```diff
@@ -0,0 +1,561 @@
+		// check for unauthorized error
+		if failureMessage, _ := status["failureMessage"].(string); failureMessage != "" {
+			if strings.Contains(failureMessage, "403") {
+				if !apimgmtv3.ClusterConditionUpdated.IsFalse(cluster) {
+					cluster = cluster.DeepCopy()
+					apimgmtv3.ClusterConditionUpdated.False(cluster)
+					apimgmtv3.ClusterConditionUpdated.Message(cluster, "cannot access EKS, check cloud credential")
+					return e.clusterClient.Update(cluster)
+				}
+			}
+		}
+
+		if !apimgmtv3.ClusterConditionUpdated.IsTrue(cluster) {
+			cluster = cluster.DeepCopy()
+			apimgmtv3.ClusterConditionUpdated.True(cluster)
+			return e.clusterClient.Update(cluster)
+		}
+	case "updating":
+		failureMessage, _ := status["failureMessage"].(string)
+
+		if !apimgmtv3.ClusterConditionProvisioned.IsTrue(cluster) {
+			// all provisioning has fnished
+			cluster = cluster.DeepCopy()
+			apimgmtv3.ClusterConditionProvisioned.True(cluster)
+			cluster, err = e.clusterClient.Update(cluster)
+			if err != nil {
+				return cluster, err
+			}
+		}
+
+		if failureMessage != apimgmtv3.ClusterConditionUpdated.GetMessage(cluster) {
+			cluster = cluster.DeepCopy()
+			apimgmtv3.ClusterConditionUpdated.Message(cluster, failureMessage)
+			if failureMessage != "" {
+				apimgmtv3.ClusterConditionUpdated.False(cluster)
+			}
+			cluster, err = e.clusterClient.Update(cluster)
+			if err != nil {
+				return cluster, fmt.Errorf("failure setting updating condition for cluster [%s]: %v", cluster.Name, err)
+			}
+		}
+
+		if !apimgmtv3.ClusterConditionUpdated.IsUnknown(cluster) && !apimgmtv3.ClusterConditionUpdated.IsFalse(cluster) {
+			cluster = cluster.DeepCopy()
+			apimgmtv3.ClusterConditionUpdated.Unknown(cluster)
+			apimgmtv3.ClusterConditionUpdated.Message(cluster, "")
+			cluster, err = e.clusterClient.Update(cluster)
+			if err != nil {
+				return cluster, fmt.Errorf("failure setting updating condition for cluster [%s] to unknown: %v", cluster.Name, err)
+			}
+		}
+
+		if failureMessage == "" {
+			logrus.Infof("waiting for cluster EKS [%s] to update", cluster.Name)
+		} else {
+			logrus.Infof("waiting for cluster EKS [%s] update failure to be resolved", cluster.Name)
+		}
+
+		e.clusterEngqueueAfter(cluster.Name, 10*time.Second)
+	default:
+		if aws.BoolValue(cluster.Spec.EKSConfig.Imported) {
+			logrus.Infof("waiting for cluster EKS [%s] to import", cluster.Name)
+			if !apimgmtv3.ClusterConditionPending.IsUnknown(cluster) || !apimgmtv3.ClusterConditionWaiting.IsUnknown(cluster) {
+				cluster = cluster.DeepCopy()
+				apimgmtv3.ClusterConditionWaiting.Unknown(cluster)
+				apimgmtv3.ClusterConditionPending.Unknown(cluster)
+				return e.clusterClient.Update(cluster)
+			}
+		} else {
+			logrus.Infof("waiting for cluster EKS [%s] to create", cluster.Name)
+		}
+		e.clusterEngqueueAfter(cluster.Name, 20*time.Second)
+		return cluster, nil
+	}
+
+	eksClusterConfigMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&cluster.Spec.EKSConfig)
```

Comment:
In what case does this section get hit ? Looks like the cases above all return.

Reply:
active and updating cluster do not return by this point unless an error was encountered. Went ahead and remove some unnecessary returns after updates for readability.